include/ethernet.h

```diff
@@ -0,0 +1,92 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2021  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_ETHERNET_H
+#define DOSBOX_ETHERNET_H
+
+#include <functional>
+
+#include "config.h"
+
+/** A virtual Ethernet connection
+ * While emulated Ethernet adapters provide the ability for the guest OS to
+ * send and receive Ethernet packets, the emulator itself needs to pass these
+ * packets to device on the host system.
+ * The EthernetConnection class provides a virtual connection for passing
+ * packets to and from a backend, intended for use by emulated adapter code.
+ */
+class EthernetConnection {
+public:
+	/** Initializes the connection.
+	 * After creating an EthernetConnection, it must have its connection
+	 * opened and initialized before further use.
+	 * This is done in a separate function instead of a constructor so any
+	 * error code may be propagated.
+	 * Do not call this function twice.
+	 * @param config The connection's configuration section
+	 * @return True if the connection is initialized, false otherwise
+	 */
+	virtual bool Initialize(Section *config) = 0;
```

Comment:
*(just a comment, not a problem)* - we don't use doxygen; you can keep commenting parameters and return types, but we prefer to have good, self-documenting parameter names instead of trivial documentation that will likely be ignored by reader.

In this case I think method name and parameter name are a-ok - but e.g. documenting destructor behaviour is rather pointless.

Reply:
Was previously dropped.